src/FSharp.Control.TaskSeq/TaskSeqInternal.fs

```diff
@@ -861,6 +869,90 @@ module internal TaskSeqInternal =
                 yield e.Current
         }
 
+    /// InsertAt or InsertManyAt
+    let insertAt index valueOrValues (source: TaskSeq<_>) =
+        if index < 0 then
+            raiseCannotBeNegative (nameof index)
+
+        taskSeq {
+            let mutable i = 0
+
+            for item in source do
+                if i = index then
+                    match valueOrValues with
+                    | Many values -> yield! values
+                    | One value -> yield value
+
+                yield item
+                i <- i + 1
+
+            // allow inserting at the end
+            if i = index then
+                match valueOrValues with
+                | Many values -> yield! values
+                | One value -> yield value
+
+            if i < index then
+                raiseOutOfBounds (nameof index)
```

Comment:
if this is a dynamic case, like this, is argument exception appropriate?
should it not be invalidOperationExceptions? IndexOutOfRangeExpcetions? 

I know I suggested that ArgumentOutOfRangeException might be better too, but it's kinda wrong for the same reasons no?

Only raising questions; I assume this is derived from what Seq does/decided...